prody/proteins/interactions.py

```diff
@@ -885,29 +900,53 @@ def calcHydrogenBondsDCD(atoms, trajectory, distA=3.0, angle=40, cutoff_dist=20,
         except TypeError:
             raise TypeError('coords must be an object '
                             'with `getCoords` method')
-        
-    if isinstance(trajectory, Atomic):
-        trajectory = Ensemble(trajectory)
-                        
+
     HBs_all = []
+
+    if trajectory is not None: 
+        if isinstance(trajectory, Atomic):
+            trajectory = Ensemble(trajectory)
+            
+        trajectory.reset()
         
-    for j0, frame0 in enumerate(trajectory):  
-        LOGGER.info('Frame: {0}'.format(j0))
-        protein = atoms.select('protein')
-        hydrogen_bonds = calcHydrogenBonds(protein, distA, angle, cutoff_dist, **kwargs)
-        HBs_all.append(hydrogen_bonds)
-        
+        for j0, frame0 in enumerate(trajectory):  
+            LOGGER.info('Frame: {0}'.format(j0))
+            protein = atoms.select('protein')
```

Comment:
This doesn't feel like it would work, but maybe I'm wrong. I don't see anywhere where you set the coordinates of the frame to be related to the atoms.

Reply:
I tested each change that I made in the code. But I will double-check.

Reply:
ok

Reply:
It is working correctly.